internal/tuplemapper/tuplemapper.go

```diff
@@ -0,0 +1,91 @@
+package tuplemapper
+
+import (
+	"context"
+	"fmt"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/tuple"
+)
+
+type Mapper[ResultType any] interface {
```

Comment:
Wondering if allow ResultType other than string makes this too generic and hard to use.  (i.e., the discussion of `New(kind MapperKind, iter storage.TupleIterator)` below)